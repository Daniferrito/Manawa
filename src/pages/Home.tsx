import { IonContent, IonHeader, IonPage, IonTitle, IonToolbar } from '@ionic/react';
import React from 'react';
import { useTranslation } from 'react-i18next';

const Home = ({ thing }: { thing?: string }) => {
  const { t } = useTranslation();
  if (thing === undefined) {
    thing = 'oyster';
  }

  return (
    <IonPage>
      <IonHeader>
        <IonToolbar>
          <IonTitle>Ionic Blank</IonTitle>
        </IonToolbar>
      </IonHeader>
      <IonContent className="ion-padding">
        {t('The world is your')} {thing}.
        <p>
          If you get lost, the{' '}
          <a target="_blank" rel="noopener noreferrer" href="https://ionicframework.com/docs/">
            docs
          </a>{' '}
          will be your guide.
        </p>
      </IonContent>
    </IonPage>
  );
};

export default Home;
