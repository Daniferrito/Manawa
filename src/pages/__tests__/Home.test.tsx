import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Home from '../Home';
import renderer from 'react-test-renderer';

afterEach(cleanup);

test('renders without crashing', () => {
  const { baseElement } = render(<Home />);
  expect(baseElement).toBeDefined();
});

test('matches the snapshots', () => {
  const component = renderer.create(<Home />);
  expect(component).toMatchSnapshot();
  const component2 = renderer.create(<Home thing="place" />);
  expect(component2).toMatchSnapshot();
});

test('sets the text to the right thing', () => {
  const component = render(<Home />);
  expect(component.queryByText(/oyster/)).toBeTruthy();
  const component2 = render(<Home thing="place" />);
  expect(component2.queryByText(/place/)).toBeTruthy();
});
