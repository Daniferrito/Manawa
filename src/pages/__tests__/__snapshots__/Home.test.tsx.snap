// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshots 1`] = `
<div
  className="ion-page"
>
  <ion-header>
    <ion-toolbar>
      <ion-title>
        Ionic Blank
      </ion-title>
    </ion-toolbar>
  </ion-header>
  <ion-content>
    The world is your
     
    oyster
    .
    <p>
      If you get lost, the
       
      <a
        href="https://ionicframework.com/docs/"
        rel="noopener noreferrer"
        target="_blank"
      >
        docs
      </a>
       
      will be your guide.
    </p>
  </ion-content>
</div>
`;

exports[`matches the snapshots 2`] = `
<div
  className="ion-page"
>
  <ion-header>
    <ion-toolbar>
      <ion-title>
        Ionic Blank
      </ion-title>
    </ion-toolbar>
  </ion-header>
  <ion-content>
    The world is your
     
    place
    .
    <p>
      If you get lost, the
       
      <a
        href="https://ionicframework.com/docs/"
        rel="noopener noreferrer"
        target="_blank"
      >
        docs
      </a>
       
      will be your guide.
    </p>
  </ion-content>
</div>
`;
